salt/utils/job.py

```diff
@@ -103,10 +103,12 @@ def store_job(opts, load, event=None, mminion=None):
         log.error(emsg)
         raise KeyError(emsg)
 
-    try:
-        mminion.returners[savefstr](load['jid'], load)
-    except KeyError as e:
-        log.error("Load does not contain 'jid': %s", e)
+    if job_cache != 'local_cache':
```

Comment:
Why is the local cache different? Trying to solve this at this level doesn't feel like the right approach.

Reply:
I would normally agree with you, but this was how this worked in the local cache _before_ the logic change in #43454. This was the least invasive way to fix this without going down many local_cache rabbit holes which could potentially break many other things.

Reply:
OK. Thanks for the explanation.